src/vm/domainfile.cpp

```diff
@@ -787,18 +782,6 @@ BOOL DomainFile::IsZapRequired()
 
     DomainAssembly * pDomainAssembly = GetDomainAssembly();
 
-    // If the manifest module does not have an ngen image, the non-manifest
-    // modules cannot either
-    if (m_pFile->IsModule() && !pDomainAssembly->GetFile()->CanUseNativeImage())
-        m_pFile->SetCannotUseNativeImage();
-
-    // Some cases are not supported by design. They can never have a native image.
-    // So ignore such cases
-
-    if (!m_pFile->CanUseNativeImage() &&
-        g_pConfig->RequireZaps() == EEConfig::REQUIRE_ZAPS_SUPPORTED)
```

Comment:
When I looked around the real reason why removing all of this `CanUseNativeImage` is that the `g_pConfig->RequireZaps() == EEConfig::REQUIRE_ZAPS_SUPPORTED` will never be true. Because the `REQUIRE_ZAPS_SUPPORTED` is not used anywhere else. So in that case, can you please also remove its definition in `eeconfig.h`.